build.sbt

```diff
@@ -10,6 +10,17 @@ scalacOptions in ThisBuild      ++= Seq("-deprecation", "-feature")
 
 cancelable in Global := true
 
+val disablePublishing = Seq[Setting[_]](
+  publishArtifact := false,
+  // The above is enough for Maven repos but it doesn't prevent publishing of ivy.xml files
+  publish := {},
+  publishLocal := {}
+)
+
+lazy val root = (project in file("."))
+  .settings(disablePublishing)
+  .aggregate(core, junit, scalacheck, testmacros)
```

Comment:
Minor/pro-tip: You can avoid having to enumerate these, as well as maintain the enumeration, by just specifying the settings at the root level:

```scala
disablePublishing
```